ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/SqlLedger.scala

```diff
@@ -425,89 +426,93 @@ private final class SqlLedgerFactory(ledgerDao: LedgerDao)(implicit logCtx: Logg
     // headRef. We also are not concerns with heartbeats / checkpoints. This is OK since this initialization
     // step happens before we start up the sql ledger at all, so it's running in isolation.
 
-    initialLedgerId match {
-      case Some(initialId) =>
-        ledgerDao
-          .lookupLedgerId()
-          .flatMap {
-            case Some(foundLedgerId) if foundLedgerId == initialId =>
-              if (initialLedgerEntries.nonEmpty) {
-                logger.warn(
-                  s"Initial ledger entries provided, presumably from scenario, but I'm picking up from an existing database, and thus they will not be used")
-              }
-              if (packages.listLfPackagesSync().nonEmpty) {
-                logger.warn(
-                  s"Initial packages provided, presumably as command line arguments, but I'm picking up from an existing database, and thus they will not be used")
-              }
-              ledgerFound(foundLedgerId)
-            case Some(foundLedgerId) =>
-              val errorMsg =
-                s"Ledger id mismatch. Ledger id given ('$initialId') is not equal to the existing one ('$foundLedgerId')!"
-              logger.error(errorMsg)
-              sys.error(errorMsg)
-            case None =>
-              if (initialLedgerEntries.nonEmpty) {
-                logger.info(
-                  s"Initializing ledger with ${initialLedgerEntries.length} ledger entries")
-              }
-
-              val contracts = acs.activeContracts.values.toList
-
-              val initialLedgerEnd = 0L
-              val entriesWithOffset = initialLedgerEntries.foldLeft(
-                (initialLedgerEnd, immutable.Seq.empty[(Long, LedgerEntry)]))((acc, le) => {
-                val offset = acc._1
-                val seq = acc._2
-                le match {
-                  case LedgerEntryOrBump.Entry(entry) =>
-                    (offset + 1, seq :+ offset -> entry)
-                  case LedgerEntryOrBump.Bump(increment) =>
-                    (offset + increment, seq)
-                }
-              })
-
-              implicit val ec: ExecutionContext = DEC
-              for {
-                _ <- doInit(initialId)
-                _ <- copyPackages(packages, timeProvider.getCurrentTime, entriesWithOffset._1)
-                _ <- ledgerDao.storeInitialState(
-                  contracts,
-                  entriesWithOffset._2,
-                  entriesWithOffset._1)
-              } yield { initialId }
-
-          }(DEC)
-
-      case None =>
-        logger.info("No ledger id given. Looking for existing ledger in database.")
-        ledgerDao
-          .lookupLedgerId()
-          .flatMap {
-            case Some(foundLedgerId) => ledgerFound(foundLedgerId)
-            case None =>
-              val randomLedgerId = LedgerIdGenerator.generateRandomId()
-              doInit(randomLedgerId).map(_ => randomLedgerId)(DEC)
-          }(DEC)
```

Comment:
Previously, this code path was never hit. Which is a good thing, because it was wrong—it didn't take into account any packages or entries provided.